packages/kbn-content-management-utils/src/saved_object_content_storage.ts

```diff
@@ -280,13 +290,28 @@ export abstract class SOContentStorage<Types extends CMCrudTypes>
       createOptions
     );
 
+    const result = {
+      item: savedObjectToItem(savedObject, this.allowedSavedObjectAttributes, false),
+    };
+
+    const validationError = transforms.create.out.result.validate(result);
+    if (validationError) {
+      if (this.throwOnResultValidationError) {
```

Comment:
This approach makes sense to me. If we're intending to throw in dev and log in prod, is the idea that we'd set `throwOnResultValidationError` from the consumer by checking `isDev`?

Reply:
Sounds good to me, I'm on board with this approach 👍